altair/examples/co2_concentration.py

```diff
@@ -0,0 +1,66 @@
+"""
+Iowa's renewable energy boom
+----------------------------
+This example is a fully developed line chart that uses a window transformation. It was inspired by `Gregor Aisch's work at datawrapper <https://www.datawrapper.de/_/OHgEm/>_`.
+"""
+# category: case studies
+import altair as alt
+from vega_datasets import data
+
+source = data.co2_concentration()
+
+base = alt.Chart(
+    source,
+    title="Carbon Dioxide in the Atmosphere"
+).transform_calculate(
+    year="year(datum.Date)"
+).transform_calculate(
+    decade="floor(datum.year / 10)"
+).transform_calculate(
+    scaled_date="(datum.year % 10) + (month(datum.Date)/12)"
+).transform_window(
+    window=[
+        {'op': 'first_value', 'as': 'first_date', 'field': 'scaled_date'},
+        {'op': 'last_value', 'as': 'last_date', 'field': 'scaled_date'}
+    ],
+    sort=[
+        {"field": "scaled_date", "order": "ascending"},
+        {"field": "scaled_date", "order": "ascending"}
+    ],
+    groupby=["decade"],
+    frame=[None, None]
```

Comment:
The window transform can be simplified a bit using the syntax available on the Python side. I think this should be equivalent:
```python
.transform_window(
    first_date='first_value(scaled_date)',
    last_date='last_value(scaled_date)',
    sort=[{"field": "scaled_date", "order": "ascending"}],
    groupby=['decade'],
    frame=[None, None]
)
```